test/open_telemetry_test.exs

```diff
@@ -32,4 +32,24 @@ defmodule OpenTelemetryTest do
       end
     end
   end
+
+  test "can deconstruct a span context" do
+    Tracer.with_span "span-1" do
+      span = Tracer.current_span_ctx()
+
+      assert nil != Span.trace_id(span)
+      assert nil != Span.span_id(span)
+      assert []   = Span.tracestate(span)
+    end
+  end
+
+  test "can establish link to parent trace" do
+    parent_trace_id = 1
+    expected_link = OpenTelemetry.link(parent_trace_id, 2, [], [])
+    Tracer.with_span "span-1", %{ links: [expected_link] } do
+      span = Tracer.current_span_ctx()
+
+      assert parent_trace_id == Span.trace_id(span)
+    end
```

Comment:
@tsloughter this is currently failing with:

```elixir
  1) test can establish link to parent trace (OpenTelemetryTest)
     test/open_telemetry_test.exs:46
     Assertion with == failed
     code:  assert parent_trace_id == Span.trace_id(span)
     left:  1
     right: 0
     stacktrace:
       test/open_telemetry_test.exs:49: (test)
```

Which makes me think that either the test is testing the wrong thing, or that `with_span/2` isn't honoring the links as I had expected it to. Considering there's no way of manually building a `OpenTelemetry.span_ctx()`, I'd expect it to treat links as such.

Reply:
I don't remember writing this test. You are right that it looks wrong. There is no reason the new span in `with_span` should have the same parent trace id as is used in the link.

Reply:
Oh wait, this is a test you added?

Reply:
probably just don't test this here, we'll want a test for it in the SDK where we can pull the links from the span to verify.